docs/virtual_machines/accessing_virtual_machines.md

```diff
@@ -290,21 +216,170 @@ spec:
+            bootcmd:
+              - setenforce 0
+        name: cloudinitdisk
+EOF
+```
+
+### Accessing SSH with virtctl
+
+Every `VirtualMachineInstance` provides a `virtualmachineinstances/portforward`
+subresource, that can be used to create a websocket backed network tunnel to
+a port inside the instance similar to Kubernetes pods.
+
+One use-case for this subresource is to forward SSH traffic into
+the `VirtualMachineInstance` either from the CLI or a web-UI.
 
-            sudo setenforce Permissive
-            sudo yum install -y qemu-guest-agent
-            sudo systemctl start qemu-guest-agent
-        name: disk1
-END
+To connect to a `VirtualMachineInstance` from your local machine, `virtctl`
+provides a lightweight SSH client with the `ssh` command, that uses port
+forwarding. Refer to the command's help for more details.
 
-kubectl create -f my-vm.yaml
+```shell
+virtctl ssh
+```
+
+To transfer files from or to a `VirtualMachineInstance` `virtctl` also
+provides a lightweight SCP client with the `scp` command. Its usage is
+similar to the `ssh` command. Refer to the command's help for more details.
+
+```shell
+virtctl scp
+```
+
+#### Using virtctl as proxy
+
+If you prefer to use your local OpenSSH client, there are two ways of doing
+that in combination with virtctl.
+
+> Note: Most of this applies to the `virtctl scp` command too.
+
+1. The `virtctl ssh` command has a `--local-ssh` option. With this option
+   `virtctl` wraps the local OpenSSH client transparently to the user. The
+   executed SSH command can be viewed by increasing the verbosity (`-v 3`).
+
+```shell
+virtctl ssh --local-ssh -v 3 testvm
+```
+
+2. The `virtctl port-forward` command provides an option to tunnel a single
+   port to your local stdout/stdin. This allows the command to be used in
+   combination with the OpenSSH client's `ProxyCommand` option.
+
+```shell
+ssh -o 'ProxyCommand=virtctl port-forward --stdio=true vmi/testvm.mynamespace 22' fedora@testvm.mynamespace
+```
+
+To provide easier access to arbitrary virtual machines you can add the following
+lines to your SSH `config`:
+
+```
+Host vmi/*
+   ProxyCommand virtctl port-forward --stdio=true %h %p
+Host vm/*
+   ProxyCommand virtctl port-forward --stdio=true %h %p
+```
+
+This allows you to simply call `ssh user@vmi/testvmi.mynamespace` and your
+SSH config and virtctl will do the rest. Using this method it becomes easy
+to set up different identities for different namespaces inside your SSH
+`config`.
 
+Note that all port forwarding traffic will be sent over the Kubernetes
+control plane. A high amount of connections and traffic can increase
+pressure on the API server. If you regularly need a high amount of connections
+and traffic consider using a dedicated Kubernetes `Service` instead.
+
+#### Example
+
+1. Create virtual machine and inject SSH public key as explained above
+
+2. Start virtual machine
+
+```shell
+virtctl start testvm
```

Comment:
If we agree to set `running: true` this is not required.

Reply:
Yes, I removed it.